packages/babel-parser/src/parser/statement.js

```diff
@@ -1471,7 +1482,11 @@ export default class StatementParser extends ExpressionParser {
 
   isExportDefaultSpecifier(): boolean {
     if (this.match(tt.name)) {
-      return this.state.value !== "async";
+      return (
+        this.state.value !== "async" &&
+        this.state.value !== "type" &&
+        this.state.value !== "interface"
+      );
```

Comment:
Here you need some lookahead, because `export type from "foo";` should still work.